fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/AddDecodeNodeForDictStringRule.java

```diff
@@ -879,6 +879,13 @@ public Boolean visitBinaryPredicate(BinaryPredicateOperator predicate, Void cont
             if (!predicate.getChild(1).isConstant()) {
                 return false;
             }
+
+            Type leftType = predicate.getChild(0).getType();
+            // These type predicates couldn't push down to storage engine
```

Comment:
```suggestion
            // These type predicates couldn't be pushed down to storage engine
```
and could you please add some explanation why?

Reply:
which is consistent with BE codes

```
struct ColumnRangeBuilder {
    template <PrimitiveType ptype>
    std::nullptr_t operator()(OlapScanConjunctsManager* cm, const SlotDescriptor* slot,
                              std::map<std::string, ColumnValueRangeType>* column_value_ranges) {
        if constexpr (ptype == TYPE_TIME || ptype == TYPE_NULL || ptype == TYPE_JSON || pt_is_float<ptype>) {
            return nullptr;
        } else {
```